libr/anal/p/anal_riscv.c

```diff
@@ -51,16 +51,15 @@ static struct riscv_opcode *get_opcode (insn_t word) {
 static int riscv_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, int len) {
 	const int no_alias = 1;
 	struct riscv_opcode *o = NULL;
-	insn_t word = 0;
+	ut64 word = 0;
```

Comment:
Dunno but risc uses 4 byte instructions, not 8..

> On 04 May 2016, at 09:35, Damien Zammit <notifications@github.com> wrote:
> 
> In libr/anal/p/anal_riscv.c:
> 
> > @@ -51,16 +51,15 @@ static struct riscv_opcode *get_opcode (insn_t word) {
> >  static int riscv_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *data, int len) {
> >  	const int no_alias = 1;
> >  	struct riscv_opcode *o = NULL;
> > -	insn_t word = 0;
> > +	ut64 word = 0;
> Ok, but why does libr/asm/arch/riscv/riscv.h say typedef uint64_t insn_t;
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
>